In [27]:
import pandas as pd
import numpy as np
from scipy import stats
df=pd.read_csv('datasets/Ashopping.csv', sep=',', encoding='CP949')

In [22]:
df.head()

고객ID  이탈여부     총_매출액  방문빈도  1회_평균매출액  할인권_사용 횟수  총_할인_금액  고객등급  구매유형  \
0     1     0   4007080    17    235711          1     5445     1     4   
1     2     1   3168400    14    226314         22   350995     2     4   
2     3     0   2680780    18    148932          6   186045     1     4   
3     4     0   5946600    17    349800          1     5195     1     4   
4     5     0  13745950    73    188301          9   246350     1     2   

   클레임접수여부  ...  매장_청결성  공간_편의성  시야_확보성  음향_적절성  안내_표지판_설명  친절성  신속성  책임성  \
0        0  ...       6       7       6       6          6    6    6    6   
1        0  ...       7       7       6       6          6    5    3    6   
2        1  ...       6       6       6       6          6    7    7    6   
3        1  ...       6       6       5       6          6    6    6    6   
4        0  ...       5       6       6       6          5    5    6    6   

   정확성  전문성  
0    6    6  
1    6    6  
2    6    7  
3    5    6  
4    5    6  

[5 rows x 39 columns]

#### 일표본 t-test  : 단일모집단에서 관심이 있는 연속형 변수의 평균 값을 특정 기준값과 비교
작년 A쇼핑 VIP 고객들의 평균 구매액은 약 7,000,000원이다.  
- H0(귀무가설)=A쇼핑 VIP 고객들의 총 매출액 평균은 7,700,000원이다.  
- H1(대립가설)=A쇼핑 VIP 고객들의 총 매출액 평균은 7,700,000원이 아니다.

In [9]:
print(df.총_매출액.mean())
print(stats.ttest_1samp(df['총_매출액'], 7700000))

7068689.76
Ttest_1sampResult(statistic=-2.8657430145836007, pvalue=0.00424758614832654)


일표본 t-test A쇼핑 VIP 고객들의 매출액 평균이 약 706만원이었다.  
t-value는 -2.865, p-value는 0.0042으로 귀무가설을 기각하여, 대립가설을 만족한다.  
작년의 매출액과 차이가 없으며 목표치를 달성하지 못했다.  
A쇼핑이 과도한 목표를 설정한 것이 아닌지 검토해보고, 경쟁사나 시장평균 성장률과 비교했을 때  
A 쇼핑의 실적이 상대적으로 낮다면 VIP 고객들의 매출액에 영향을 미치는 원인을 조사하여 매출확대 대책을 마련해야한다.

#### 독립표본 t-test : 두개의 독립된 모집단의 평균을 비교
- H0(귀무가설)=A쇼핑 클레임 고객들과 비클레임 고객들의 방문횟수 차이는 없다.  
- H1(대립가설)=A쇼핑 클레임 고객들과 비클레임 고객들의 방문횟수 차이는 있다.  

In [16]:
# 등분산성 검정
no_claim=df[df.클레임접수여부==0]
claim=df[df.클레임접수여부==1]
df2=np.array(no_claim.방문빈도)
df3=np.array(claim.방문빈도)
stats.bartlett(df2, df3)

BartlettResult(statistic=13.626177910965525, pvalue=0.00022305349806448475)

F값은 13.6, p-value는 0.0002로 귀무가설을 기각, 즉 두 집단의 분산은 동일하지 않다.  
등분산이 가정되지 않은 t 통계량을 확인하여 가설검증을 수행한다.

In [30]:
print(stats.ttest_ind(df2, df3, equal_var=False)) 
print('no클레임 고객 평균 방문빈도 :', no_claim.방문빈도.mean())
print('클레임 고객 평균 방문빈도 :', claim.방문빈도.mean())

Ttest_indResult(statistic=2.595726838875684, pvalue=0.009577734932789503)
no클레임 고객 평균 방문빈도 : 28.184842883548985
클레임 고객 평균 방문빈도 : 24.736383442265794


독립표본 t-value는 2.595, p-value는 0.0096로 나왔다.  
no클레임 고객은 28.2회, 클레임고객은 24.7회 평균 방문하였다.  
귀무가설이 기각되어 클레임 접수여부에 따라 방문 빈도는 차이가 있다고 할 수 있다.  
클레임 접수 고객이 VIP일 경우 클레임 접수 후 구매 빈도나 방문 횟수의 감소를 확인하여 선제적인 대응이 필요하다.

#### 대응표본 t-test  : 단일 모집단에 대해 두 번의 처리를 가했을 때, 처리에 따른 평균의 차이
 - H0(귀무가설)=A쇼핑 고객들의 멤버십 프로그램 도입 전 만족도와 도입 후 만족도의 차이는 없다.  
 - H1(대립가설)=A쇼핑 고객들의 멤버십 프로그램 도입 전 만족도와 도입 후 만족도의 차이는 있다.

In [32]:
stats.ttest_rel(df['멤버쉽_프로그램_가입후_만족도'],df['멤버쉽_프로그램_가입전_만족도'])

Ttest_relResult(statistic=29.560410783358122, pvalue=1.7319140513197275e-138)

t값은 29.560, p-value는 0.0001 이하로 나왔다.  
귀무가설을 기각하므로 멤버십 프로그램 도입전과 도입후의 만족도 차이는 있다고 할 수 있다.  
단순 포인트 중심의 멤버십보다 고객 구매가 증대될 수록 받는 사은품, 쿠폰이 커지는 새로운 멤버십이 더 효과적이다.  